File: c++_solutions/950.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        sort(deck.begin() , deck.end());
        vector<int> new_deck(deck.size());
        queue<int>ind ; 
        for(size_t idx = 0 ; idx < deck.size() ; idx++){
            ind.push(idx) ;
        }
        size_t idx = 0 ;
        while(idx < deck.size()){
            new_deck[ind.front()] = deck[idx] ; 
            idx++;
            ind.pop();
            if(ind.size() != 0){
                auto elm = ind.front();
                ind.pop();
                ind.push(elm);
            }
        }
        return new_deck ; 
    }
};

File: rust_solutions/950.rs
impl Solution {
    pub fn deck_revealed_increasing(deck: Vec<i32>) -> Vec<i32> {
        let mut prev_deck = deck.clone() ; 
        use std::collections::LinkedList ; 
        let mut ind = LinkedList::new() ; 
        for idx in 0..deck.len() {
            ind.push_back(idx);
        }
        prev_deck.sort_unstable() ; 
        let mut new_deck = vec![0;deck.len()] ;
        for num in prev_deck.iter() {
            let idx = match ind.pop_front() {
                Some(idx) => idx ,
                _ => 0
            };
            new_deck[idx] = *num ; 
            if let Some(idx) = ind.pop_front() {
                ind.push_back(idx) ;
            }
        }
        
        new_deck
    }
}
